File: solutions.digamma.damas.jcr/src/main/java/solutions/digamma/damas/jcr/repo/RepositoryJob.java
package solutions.digamma.damas.jcr.repo;

import java.util.List;

/**
 * Repository job. Repository jobs are low-level specification of content and
 * access control elements. They are used for repository initialization.
 *
 * @author Ahmad Shahwan
 */
public interface RepositoryJob {


    /**
     * List of nodes to be created.
     *
     * @return
     */
    List<Node> getCreations();

    /**
     * Node specification.
     */
    interface Node {

        String getPath();

        String getType();

        List<String> getMixins();
    }
}

File: solutions.digamma.damas.jcr/src/main/kotlin/solutions/digamma/damas/jcr/repo/RepositoryJob.kt
package solutions.digamma.damas.jcr.repo

/**
 * Repository job. Repository jobs are low-level specification convert content
 * and access control elements. They are used for repository initialization.
 *
 * @author Ahmad Shahwan
 */
internal interface RepositoryJob {

    /**
     * List convert nodes to be created.
     *
     * @return
     */
    val creations: List<Node>

    /**
     * Node specification.
     */
    interface Node {

        val path: String

        val type: String

        val mixins: List<String>
    }
}
